octodns/provider/ns1.py

```diff
@@ -275,49 +545,342 @@ def populate(self, zone, target=False, lenient=False):
+            if feed_id:
+                self._client.datafeed_delete(self._client.datasource_id,
+                                             feed_id)
+
+            self._client.monitors_delete(monitor_id)
+
+            notify_list_id = monitor['notify_list']
+            self._client.notifylists_delete(notify_list_id)
+
+    def _params_for_dynamic_A(self, record):
+        pools = record.dynamic.pools
+
+        # Convert rules to regions
+        regions = {}
+        for i, rule in enumerate(record.dynamic.rules):
+            pool_name = rule.data['pool']
+
+            notes = {
+                'rule-order': i,
+            }
+
+            fallback = pools[pool_name].data.get('fallback', None)
+            if fallback:
+                notes['fallback'] = fallback
+
+            country = set()
+            georegion = set()
+            us_state = set()
+
+            for geo in rule.data.get('geos', []):
+                n = len(geo)
+                if n == 8:
+                    # US state
+                    us_state.add(geo[-2:])
+                elif n == 5:
+                    # Country
+                    country.add(geo[-2:])
+                else:
+                    # Continent
+                    georegion.update(self._CONTINENT_TO_REGIONS[geo])
+
+            meta = {
+                'note': self._encode_notes(notes),
+            }
+            if georegion:
+                meta['georegion'] = sorted(georegion)
+            if country:
+                meta['country'] = sorted(country)
+            if us_state:
+                meta['us_state'] = sorted(us_state)
+
+            regions[pool_name] = {
+                'meta': meta,
+            }
+
+        existing_monitors = self._monitors_for(record)
+        active_monitors = set()
+
+        # Build a list of primary values for each pool, including their
+        # feed_id (monitor)
+        pool_answers = defaultdict(list)
+        for pool_name, pool in sorted(pools.items()):
+            for value in pool.data['values']:
+                weight = value['weight']
+                value = value['value']
+                existing = existing_monitors.get(value)
+                monitor_id, feed_id = self._monitor_sync(record, value,
+                                                         existing)
+                active_monitors.add(monitor_id)
+                pool_answers[pool_name].append({
+                    'answer': [value],
+                    'weight': weight,
+                    'feed_id': feed_id,
+                })
+
+        default_answers = [{
+            'answer': [v],
+            'weight': 1,
+        } for v in record.values]
+
+        # Build our list of answers
+        answers = []
+        for pool_name in sorted(pools.keys()):
+            priority = 1
+
+            # Dynamic/health checked
+            current_pool_name = pool_name
+            while current_pool_name:
```

Comment:
Usual caveat that it would be best to have an infinite loop detection here just to be better safe than sorry.

Reply:
Added something simple here, but fwiw it's not really possible to create a `Record` with a fallback loop due to https://github.com/github/octodns/blob/208e2b36168cbb7e3972a69de82a5a28d836dc1f/octodns/record/__init__.py#L554-L566.